Lib/test/libregrtest/single.py

```diff
@@ -195,7 +203,7 @@ def _runtest(result: TestResult, runtests: RunTests) -> None:
         timeout is not None and threading_helper.can_start_thread
     )
     if use_timeout:
-        faulthandler.dump_traceback_later(timeout, exit=True)
+        faulthandler.dump_traceback_later(cast(float, timeout), exit=True)
```

Comment:
I don't understand this cast. timeout is a float is use_timeout is True. Why is a cast needed?

Reply:
Yes, this code is type-safe. The issue is that mypy only understands a [limited set of type narrowing operations](https://mypy.readthedocs.io/en/stable/type_narrowing.html#type-narrowing-expressions), and the way `timeout` has been narrowed to be a `float` here isn't one of them.

Mypy understands this kind of logic:

```py
x: float | None
if x is not None:
    # OK, mypy understands `f` has to be a float here
```

But this kind of thing is too complex for mypy:

```py
x: float | None
x_is_None = x is None
if not x_is_None:
    # mypy will not understand that x has been safely narrowed to `None` here
```

Another way we could make mypy happy would be to add an `assert`, like this -- but it feels redundant to me, since the type of `timeout` has already been narrowed to be a `float` in the lines immediately preceding this:

```suggestion
        assert timeout is not None
        faulthandler.dump_traceback_later(timeout, exit=True)
```